azext_iot/tests/iothub/message_endpoint/test_iothub_message_endpoint_unit.py

```diff
@@ -0,0 +1,938 @@
+
+hub_name = "hubname"
+hub_rg = "hubrg"
+endpoint_name = generate_names()
+resource_not_found_error = "Resource not found."
+generic_response = {generate_names(): generate_names()}
+
+
+path_find_resource = "azext_iot.iothub.providers.discovery.IotHubDiscovery.find_resource"
+arg_check_path = "azext_iot.iothub.providers.message_endpoint.MessageEndpoint._connection_string_retrieval_args_check"
+get_eventhub_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_eventhub_cstring"
+get_servicebus_topic_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_servicebus_topic_cstring"
+get_servicebus_queue_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_servicebus_queue_cstring"
+get_cosmos_db_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_cosmos_db_cstring"
+parse_cosmos_db_cstring_path = "azext_iot.iothub.providers.message_endpoint.parse_cosmos_db_connection_string"
+get_storage_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_storage_cstring"
+
+
+@pytest.fixture()
+def fixture_update_endpoint_ops(mocker):
+    # Fetch connection string
+    arg_check = mocker.patch(arg_check_path)
+    mocker.patch(get_eventhub_cstring_path, return_value="get_eventhub_cstring")
+    mocker.patch(get_servicebus_topic_cstring_path, return_value="get_servicebus_topic_cstring")
+    mocker.patch(get_servicebus_queue_cstring_path, return_value="get_servicebus_queue_cstring")
+    mocker.patch(get_storage_cstring_path, return_value="get_storage_cstring")
+    mocker.patch(get_cosmos_db_cstring_path, return_value="get_cosmos_db_cstring")
+    mocker.patch(parse_cosmos_db_cstring_path, return_value={
+        "AccountKey": "get_cosmos_db_account_key",
+        "AccountEndpoint": "get_cosmos_db_account_endpoint"
+    })
+
+    # Hub Resource
+    find_resource = mocker.patch(path_find_resource, autospec=True)
+
+    def create_mock_endpoint():
+        endpoint = mocker.Mock()
+        endpoint.name = endpoint_name
+        return endpoint
+
+    hub_mock = mocker.MagicMock()
+    hub_mock.properties.routing.endpoints.event_hubs = [create_mock_endpoint()]
+    hub_mock.properties.routing.endpoints.service_bus_queues = [create_mock_endpoint()]
+    hub_mock.properties.routing.endpoints.service_bus_topics = [create_mock_endpoint()]
+    hub_mock.properties.routing.endpoints.storage_containers = [create_mock_endpoint()]
+    hub_mock.properties.routing.endpoints.cosmos_db_sql_collections = [create_mock_endpoint()]
+
+    def initialize_mock_client(self, *args):
+        self.client = mocker.MagicMock()
+        self.client.begin_create_or_update.return_value = generic_response
+        return hub_mock
+
+    find_resource.side_effect = initialize_mock_client
+
+    yield find_resource, arg_check
+
+
+class TestMessageEndpointUpdate:
+    @pytest.mark.parametrize(
+        "req",
+        [
+            {},
+            {
+                "endpoint_account_name": None,
```

Comment:
Should I keep the key/value pairs where the value is None? It helps with readability but does make it so there is more stuff to read

Reply:
I think either way is fine - either explicitly set them to `None`, or don't set them, and then check that their values are the same as before (either non-existent or `None`) - am I understanding that correctly?

Reply:
if they are not present, it is the same as if they are None (or the default which is None)

In the tests, I am checking if a param was set by seeing if the mock object has it or not (if it doesnt, trying to call that attribute will cause it to give a mock object)